clients/wallet/src/horizon/responses.rs

```diff
@@ -471,29 +473,54 @@ impl<C: HorizonClient> TransactionsResponseIter<C> {
 	}
 
 	#[doc(hidden)]
-	// returns the first record of the list
+	// returns the first transaction response of the list
 	fn get_top_record(&mut self) -> Option<TransactionResponse> {
 		if !self.is_empty() {
 			return Some(self.records.remove(0))
 		}
 		None
 	}
 
-	/// returns the next TransactionResponse in the list
+	/// returns the next TransactionResponse
 	pub async fn next(&mut self) -> Option<TransactionResponse> {
 		match self.get_top_record() {
 			Some(record) => Some(record),
 			None => {
-				// call the next page
-				tracing::trace!("calling next page: {}", &self.next_page);
-
-				let response: HorizonTransactionsResponse =
-					self.client.get_from_url(&self.next_page).await.ok()?;
-				self.next_page = response.next_page();
-				self.records = response.records();
-
+				let _ = self.jump_to_next_page().await?;
 				self.get_top_record()
 			},
 		}
 	}
+
+	/// returns the TransactionResponse in the middle of the list
+	pub fn middle(&mut self) -> Option<TransactionResponse> {
```

Comment:
this is important for https://github.com/pendulum-chain/spacewalk/blob/bda384bcc6db3da88198c714851a261dcdea1063/clients/wallet/src/resubmissions.rs#L250